### ISSUE TITLE
hunter-trace my-script.py args-for-my-script

### ISSUE THREAD

**Issue by guettli** [ionelmc/python-hunter#48]
I would like to call hunter-trace like this:

    hunter-trace my-script.py args-for-my-script

Is there a reason why this is not implemented up to now?

**Comment by ionelmc** [comment]
Not implemented cause this is supported via env var activation. Your example is equivalent to `PYTHONHUNTER= my-script.py args-for-my-script` in bash parlance. On windows you'd do a `set PYTHONHUNTER=" "` before running the script.

**Comment by guettli** [comment]
I think convenience is important.

It would be great to have a simple command line tool which accepts the arguments like the python trace modules accepts: https://docs.python.org/2/library/trace.html#filters

* --ignore-module=<mod>
*  --ignore-dir=<dir>

This would make it easier for new comers.

**Comment by ionelmc** [comment]
How exactly replicating the insane quirks of stdlib trace module would make hunter a better tracer? That's exactly what I wanted to avoid.

If you want a drop-in replacement why not make your own wrapper?

**Comment by guettli** [comment]
Yes, you are right. Insane quirks of stdlib trace a not nice.

Yes, I could write a wrapper. But I think it is better to improve the upstream. In this case python-hunter.

I don't need a drop-in replacement. But these two command line arguments are easy and straightforwards.

Would you accept a pull request for a wrapper?

**Comment by ionelmc** [comment]
Maybe, but I'd like to understand your actual needs first. I suspect your problem is that the predicate mumbojumbo ain't straightforward enough. You'd rather type `--ignore-module=foobar` rather than `~Q(module='foobar')`. Is the `~` operator unclear? Would ``Not(Q(module='foobar'))` be better?

How about `hunter-trace --help` giving a short intro to filtering and such?

Also, what I don't like about `python -mtrace` is that it only works with scripts. If you got a module that you'd like to run then tough luck - `python -mtrace -mfoobar` aint't gonna work. If you have a bin somewhere (from a `console_scripts` entrypoint or whatever) then `python -mtrace foobar` ain't gonna work. You end up with `python -mtrace $(which foobar)` or some other shell contraption. Also, you might end up using the wrong interpreter.

**Comment by guettli** [comment]
Yes,  "predicate mumbojumbo ain't straightforward enough."

A command which you write before the real command is well known at least in the linux world. Like `nice` or `strace`. It is obvious.

I thought `hunter-trace` is exactly this. But know I learned.

I just wanted to let you know what the predicate mumbojumbo is not obvious for everyone.

But it works.

**Comment by abitrolly** [comment]
I agree that `PYTHONHUNTER=` is convenient for debugging `python -m <module>` and functions within modules without modifying the modules. However, I also find it convenient to invoke `hunter script.py` without prior knowledge about `pid` or `PYTHONHUNTER=` env variable together with a `.pth` mechanism of injecting code into Python interpreter.
